graphix/_db.py

```diff
@@ -2,43 +2,51 @@
 
 from __future__ import annotations
 
+from typing import ClassVar, TypeVar
+
 import numpy as np
 import numpy.typing as npt
 
+_T = TypeVar("_T", bound=np.generic)
+
 
-def _lock(m: npt.NDArray[np.complex128]) -> npt.NDArray[np.complex128]:
-    """Create a true immutable view."""
+def _lock(data: npt.NDArray[_T]) -> npt.NDArray[np.complex128]:
+    """Create a true immutable view.
+
+    data must be a "temporary" array, i.e., should not be bound to a variable.
+    """
+    m = data.astype(np.complex128)
```

Comment:
I added `.astype` here for simpler codes.
Since we cannot omit `[...]` in `npt.NDArray` (mypy strict rule), we would end up casting array literal to `np.complex128` somewhere else if we were to omit the cast here.